### PR TITLE
Fix serialization of x-kubernetes-validations OpenAPI extension

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind bug
/kind flake

#### What this PR does / why we need it:

Switches the openapi serialized extension for `x-kubernetes-validations` to use the external type with correct json tags. It was getting serialized incorrectly as `[{"Message":"...", "Rule":"..."}]`

Makes the jsonschemaprops -> openapi -> jsonschemaprops roundtrip test more strict
* fails decoding on unrecognized fields (would have caught https://github.com/kubernetes/kubernetes/issues/107954#issuecomment-1030654083)
* seeds from the current time and increases fuzz iterations for better coverage (bugs crept in with a fixed seed and 20 iterations)

```release-note
Fix OpenAPI serialization of the x-kubernetes-validations field
```

/cc @benluddy @sttts 
/sig api-machinery

### CODE REVIEW

**Comment by liggitt** [comment]
serialization issue seen here:

```
[liggitt@liggitt-macbookpro2 kubernetes]$ TEST_RAND_SEED=389141000 go test k8s.io/apiextensions-apiserver/pkg/apiserver/validation
--- FAIL: TestRoundTrip (0.00s)
    validation_test.go:63: using overridden seed: 389141000
    validation_test.go:105: [unknown field "not.patternProperties.彤Ž叩ǖ搥Õ軯夞ȗ@.x-kubernetes-validations[0].Message" unknown field "not.patternProperties.彤Ž叩ǖ搥Õ軯夞ȗ@.x-kubernetes-validations[0].Rule"]
FAIL
FAIL	k8s.io/apiextensions-apiserver/pkg/apiserver/validation	0.886s
``

**Comment by liggitt** [comment]
This is producing an unserializable extension:

https://github.com/kubernetes/kubernetes/blob/9b09612d1b0a6ec441833f928cbb205d936da59c/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go#L257-L259

Because the referenced `XValidations` field doesn't set json tags on its rule and message fields.

Options:
1. add json tags to the internal XValidations `Rule` and `Message` fields... not great because this is an internal type
2. change ConvertJSONSchemaPropsWithPostProcess to use the v1 CRD types which *are* serializable
3. translate the XValidations field to a struct private to ConvertJSONSchemaPropsWithPostProcess that is serializable

My instinct is to go with option 2

**Comment by liggitt** [comment]
> Options:
> 
> 1. add json tags to the internal XValidations `Rule` and `Message` fields... not great because this is an internal type
> 2. change ConvertJSONSchemaPropsWithPostProcess to use the v1 CRD types which _are_ serializable
> 3. translate the XValidations field to a struct private to ConvertJSONSchemaPropsWithPostProcess that is serializable
> 
> My instinct is to go with option 2

hmm... switching conversion entirely to use v1 schema types wreaks havoc with the handling of raw JSON values for defaults, etc

**Comment by liggitt** [comment]
I defined an internal->v1 conversion for the ValidationRules type, and use the v1 type when constructing a structural schema

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/conversion_test.go [hunk 1]
```diff
@@ -660,3 +661,71 @@ func TestJSONRoundTrip(t *testing.T) {
...
+		assertEqualTypes(t, path, aElemType, bElemType)
+
+	default:
+		fatalTypeError(t, path, a, b, "unhandled kind")
+	}
+}
+
+func fatalTypeError(t *testing.T, path []string, a, b reflect.Type, message string) {
+	t.Helper()
+	t.Fatalf("%s: %s: %s %s", strings.Join(path, ""), message, a, b)
```
Comment (benluddy) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/conversion_test.go:730]: Nit ([same here](https://github.com/kubernetes/kubernetes/blob/3a132bd20694afce993efcc35b22b874e990dc4d/pkg/kubelet/cri/remote/conversion_test.go#L124)): 

```suggestion
	t.Fatalf("%s: %s: %s %s", strings.Join(path, "."), message, a, b)
```

**Comment by liggitt** [comment]
/assign @jpbetz 
for org member lgtm

**Comment by jpbetz** [comment]
/lgtm

Wow, I'm surprised we didn't notice this till now. Thanks for fixing!

**Pull request merged**